src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,334 @@
+            }
 
-		private LocationManager InitializeLocationProvider(double desiredAccuracy)
-		{
-			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+            public void OnProviderEnabled(string provider)
+            {
+                
+            }
 
-			var criteriaForLocationService = new Criteria
-			{
-				Accuracy = Accuracy.Coarse
-			};
+            public void OnStatusChanged(string provider, Availability status, Bundle extras)
+            {
+                // This method was deprecated in API level 29 (Android 10). This callback will never be invoked.
+            }
+        }
 
-			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
 
-			if (acceptableLocationProviders.Any())
-			{
-				_locationProvider = acceptableLocationProviders.First();
-			}
-			else
-			{
-				_locationProvider = String.Empty;
-			}
+        MyLocListen myListener = new MyLocListen();
+        #endregion 
 
-			return locationManager;
-		}
+        public Geolocator()
+        {
+            _locationManager = (LocationManager)Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
 
-		partial void StartPositionChanged()
-		{
+            _reportInterval = 1000;
+            _movementThreshold = 0;
+
+            RequestUpdates();
+        }
+
+        ~Geolocator()
+        {
+            try
+            {
+                RemoveUpdates();
+            }
+            catch
+            { }
+
+            try
+            {
+                _locationManager.Dispose();
+            }
+            catch
+            { }
+        }
+
+        public Task<Windows.Devices.Geolocation.Geoposition> GetGeopositionAsync()
+            => GetGeopositionAsync(TimeSpan.FromHours(2), TimeSpan.FromSeconds(60));
+
+        public async Task<Windows.Devices.Geolocation.Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
+        {
+            var providers = _locationManager.GetProviders(_locationCriteria,true);
+            int bestAccuracy = 10000;
+            Location bestLocation = null;
+			
 			BroadcastStatus(PositionStatus.Initializing);
-		}
 
-		public void OnLocationChanged(Location location)
-		{
-			BroadcastStatus(PositionStatus.Ready);
-			this._positionChanged?.Invoke(this, new PositionChangedEventArgs(location.ToGeoPosition()));
-		}
+            foreach (string locationProvider in providers)
+            {
+                var location = _locationManager.GetLastKnownLocation(locationProvider);
+                if (location != null)
+                {
+                    // check how old is this fix
+                    DateTimeOffset date = DateTimeOffset.FromUnixTimeMilliseconds(location.Time); 
+                    if (date.AddMilliseconds(maximumAge.TotalMilliseconds) > DateTimeOffset.UtcNow)
```

Comment:
```suggestion
                    if (date + maximumAge > now)
```
... and you should _capture_ `now` before the loop.